eap_backend/eap_api/views.py

```diff
@@ -22,225 +24,120 @@
-        includes the whole item tree, subtrees when recursing.
-    obj_type: Key of the json object (also a key of 'TYPE_DICT'). At the top
-        level this should be "assurance_case".
-    parent_id: None at the top level, id of the caller when recursing.
-
-    Returns
-    =======
-    objs: JsonResponse describing failure/success.
-    """
-    # Create the top object in data. Only include some of the fields from data,
-    # so that e.g. the new object gets a unique ID even if `data` specifies an
-    # ID.
-    this_data = {k: data[k] for k in TYPE_DICT[obj_type]["fields"]}
-    if parent_id is not None and parent_type is not None:
-        for parent_type_tmp, plural in TYPE_DICT[obj_type]["parent_types"]:
-            # TODO This is silly. It's all because some parent_type names are written
-            # with a plural s in the end while others are not.
-            if (
-                parent_type not in parent_type_tmp
-                and parent_type_tmp not in parent_type
-            ):
-                continue
-            if plural:
-                parent_id = [parent_id]
-            this_data[parent_type_tmp + "_id"] = parent_id
-    serializer_class = TYPE_DICT[obj_type]["serializer"]
-    serializer = serializer_class(data=this_data)
-    if serializer.is_valid():
-        serializer.save()
-    else:
+@csrf_exempt
+def user_list(request):
+    """
+    List all users, or make a new user
+    """
+    if request.method == "GET":
+        users = EAPUser.objects.all()
+        serializer = EAPUserSerializer(users, many=True)
+        return JsonResponse(serializer.data, safe=False)
+    elif request.method == "POST":
+        data = JSONParser().parse(request)
+        serializer = EAPUserSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return JsonResponse(serializer.data, status=201)
         return JsonResponse(serializer.errors, status=400)
 
-    # Recurse into child types.
-    name = serializer.data["name"]
-    id = serializer.data["id"]
-    success_http_code = 201
-    child_types = TYPE_DICT[obj_type]["children"]
-    for child_type in child_types:
-        if child_type not in data:
-            continue
-        for child_data in data[child_type]:
-            retval = save_json_tree(
-                child_data, child_type, parent_id=id, parent_type=obj_type
-            )
-            # If one of the subcalls returns an error, return.
-            if retval.status_code != success_http_code:
-                return retval
 
-    summary = {"name": name, "id": id}
-    return JsonResponse(summary, status=success_http_code)
+@csrf_exempt
+def user_detail(request, pk):
+    """
+    Retrieve, update, or delete a User by primary key
+    """
+    try:
+        user = EAPUser.objects.get(pk=pk)
+    except EAPUser.DoesNotExist:
+        return HttpResponse(status=404)
+    if request.method == "GET":
```

Comment:
Do this and the below PUT and DELETE allow seeing/modifying e.g. other users' passwords, or deleting other users?

Reply:
What if we just checked that `user == request.user` for `user_detail`? Would that block some use case we would need?

Reply:
Sure, we could do that, and it wouldn't block any use-cases I can think of...   Are you thinking we could have a user details page on the frontend?    I could definitely imagine this being something that could be useful at some point, so people can see what groups they belong to, and/or have a sort of "homepage" where they can see all their cases and groups....   (but maybe that's for a future iteration...)

Reply:
Yes, I could imagine it being useful in the future, but no, I wasn't planning on building that now. I didn't think about this from a very practical point of view, there's just a semi-compulsive part of me that feels uncomfortable exposing API end points that could be exploited, even if a realistic threat model doesn't include anyone exploiting them.